ee/codegen/src/generators/workflow.ts

```diff
@@ -716,96 +716,111 @@ export class Workflow {
         continue;
       }
 
-      if (graphMutableAst.type === "empty") {
-        graphMutableAst = {
-          type: "node_reference",
-          reference: targetNode,
-        };
-      } else if (graphMutableAst.type === "node_reference") {
-        if (sourceNode && graphMutableAst.reference === sourceNode) {
-          const sourceNodePortContext = sourceNode.portContextsById.get(
-            edge.sourceHandleId
-          );
-          if (sourceNodePortContext) {
-            if (sourceNodePortContext.isDefault) {
-              graphMutableAst = {
-                type: "right_shift",
-                lhs: graphMutableAst,
-                rhs: { type: "node_reference", reference: targetNode },
-              };
-            } else {
-              graphMutableAst = {
-                type: "right_shift",
-                lhs: {
-                  type: "port_reference",
-                  reference: sourceNodePortContext,
-                },
-                rhs: { type: "node_reference", reference: targetNode },
-              };
-            }
-          }
-        } else {
-          graphMutableAst = {
-            type: "set",
-            values: [
-              graphMutableAst,
-              { type: "node_reference", reference: targetNode },
-            ],
-          };
-        }
-      } else if (graphMutableAst.type === "set") {
-        if (
-          graphMutableAst.values.some(
-            (value) =>
-              value.type === "node_reference" && value.reference === sourceNode
-          )
-        ) {
-          graphMutableAst = {
-            type: "right_shift",
-            lhs: graphMutableAst,
-            rhs: { type: "node_reference", reference: targetNode },
+      const addEdgeToGraph = (
+        mutableAst: GraphMutableAst
+      ): GraphMutableAst | undefined => {
+        if (mutableAst.type === "empty") {
+          return {
+            type: "node_reference",
+            reference: targetNode,
           };
-        } else {
-          graphMutableAst = {
-            type: "set",
-            values: [
-              graphMutableAst,
-              { type: "node_reference", reference: targetNode },
-            ],
-          };
-        }
-      } else if (graphMutableAst.type === "right_shift") {
-        if (graphMutableAst.lhs.type == "port_reference") {
-          const sourcePortContext = sourceNode?.portContextsById.get(
-            edge.sourceHandleId
-          );
-          if (sourcePortContext) {
-            graphMutableAst = {
-              type: "set",
-              values: [
-                graphMutableAst,
-                {
+        } else if (mutableAst.type === "node_reference") {
+          if (sourceNode && mutableAst.reference === sourceNode) {
```

Comment:
renaming the function scoped arg from `graphMutableAst` to `mutableAst` just to deconflict with the variable in the while loop scope that we are updating with each edge